### PR TITLE
Update Release Drafter and Jenkinsfile with Docker credentials and promotions

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Update Release Drafter and Jenkinsfile with Docker credentials and promotions

### Issues Resolved
Closes https://github.com/opensearch-project/opensearch-build/issues/4857

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/release.jenkinsFile
Comment (peternied) [inline jenkins/release.jenkinsFile:86]: What do you think of this refactor [1] to make it easier to confirm what is being pushed where? 

- [1] https://github.com/peterzhuamazon/opensearch-migrations/pull/1

**Review by peternied — COMMENTED** [review]
Thanks for doing this @peterzhuamazon overall approach looks good, but I'd prefer if we can reduce the redundant build job definitions seems easy for a copypasta error to be hidden in there.

Maybe this PR [1] could do that - or let me know what your think about this.

- [1] https://github.com/peterzhuamazon/opensearch-migrations/pull/1

File: jenkins/release.jenkinsFile
Comment (peterzhuamazon) [reply jenkins/release.jenkinsFile]: Thanks @peternied.

Comment (peterzhuamazon) [reply jenkins/release.jenkinsFile]: Approved.

File: jenkins/release.jenkinsFile
Comment (AndreKurait) [inline jenkins/release.jenkinsFile:110]: Can we keep this commented until after our security review
```suggestion
                        // def copyDockerImage = {imageName ->
                        //     def registries = ['opensearchproject', 'public.ecr.aws/opensearchproject']
                        //     destRegistries.each { destRegistry ->
                        //         build job: 'docker-copy',
                        //         parameters: [
                        //             string(name: 'SOURCE_IMAGE_REGISTRY', value: sourceRegistry),
                        //             string(name: 'SOURCE_IMAGE', value: sourceImage),
                        //             string(name: 'DESTINATION_IMAGE_REGISTRY', value: destRegistry),
                        //             string(name: 'DESTINATION_IMAGE', value: "${imageName}:${tag}")
                        //         ]

                        //         build job: 'docker-copy',
                        //         parameters: [
                        //             string(name: 'SOURCE_IMAGE_REGISTRY', value: sourceRegistry),
                        //             string(name: 'SOURCE_IMAGE', value: sourceImage),
                        //             string(name: 'DESTINATION_IMAGE_REGISTRY', value: destRegistry),
                        //             string(name: 'DESTINATION_IMAGE', value: "${imageName}:latest")
                        //         ]
                        //     }
                        // }

                        // copyDockerImage('opensearch-migrations-console');
                        // copyDockerImage('opensearch-migrations-traffic-replayer');
                        // copyDockerImage('opensearch-migrations-traffic-capture-proxy');
                        // copyDockerImage('opensearch-migrations-reindex-from-snapshot');
```

**Review by peternied — APPROVED** [review]

**Comment by peternied** [comment]
@peterzhuamazon Thanks for this change, we will merge as soon as CI passes - have a great weekend!

**Review by AndreKurait — CHANGES_REQUESTED** [review]
LGTM, let's hold off merging until @sumobrian chimes in and we have bandwidth to test this without blocking a necessary release

**Review by AndreKurait — APPROVED** [review]

**Review by mikaylathompson — APPROVED** [review]

**Pull request merged**
